js/src/logger.ts

```diff
@@ -2924,6 +3131,78 @@ function validateAndSanitizeExperimentLogPartialArgs(
   }
 }
 
+/**
+ * Creates a deep copy of the given event. Replaces references to user objects
+ * with placeholder strings to ensure serializability, except for
+ * {@link Attachment} objects, which are preserved.
+ */
+function deepCopyEvent<T extends Partial<BackgroundLogEvent>>(event: T): T {
```

Comment:
Could we add a simple unit test for this? Should just need to be in `logger.test.ts` or something.

Reply:
Done